File: frontend/src/services/taskService.js
import axios from 'axios';

export const taskService = {
  // GET /api/tasks
  async getAllTasks() {
    const response = await axios.get('/api/tasks');
    return response.data;
  },

  // GET /api/tasks/{id}
  async getTaskById(id) {
    const response = await axios.get(`/api/tasks/${id}`);
    return response.data;
  },

  // GET /api/tasks/employee/{employeeId}
  async getTasksByEmployee(employeeId) {
    const response = await axios.get(`/api/tasks/employee/${employeeId}`);
    return response.data;
  },

  // POST /api/tasks
  async createTask(taskData) {
    const response = await axios.post('/api/tasks', taskData);
    return response.data;
  },

  // PUT /api/tasks/{id}
  async updateTask(id, taskData) {
    const response = await axios.put(`/api/tasks/${id}`, taskData);
    return response.data;
  },

  // PATCH /api/tasks/{id}/status
  async updateTaskStatus(id, status) {
    const response = await axios.patch(`/api/tasks/${id}/status`, { status });
    return response.data;
  },

  // DELETE /api/tasks/{id}
  async deleteTask(id) {
    await axios.delete(`/api/tasks/${id}`);
    return true;
  }
};

File: backend/src/main/java/com/taskmanager/service/TaskService.java
package com.taskmanager.service;

import com.taskmanager.dto.TaskRequest;
import com.taskmanager.exception.ResourceNotFoundException;
import com.taskmanager.model.Task;
import com.taskmanager.model.TaskStatus;
import com.taskmanager.repository.TaskRepository;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class TaskService {

    private final TaskRepository taskRepository;

    public TaskService(TaskRepository taskRepository) {
        this.taskRepository = taskRepository;
    }

    public Task createTask(TaskRequest request) {
        Task task = new Task();
        task.setTitle(request.getTitle());
        task.setDescription(request.getDescription());
        task.setStatus(request.getStatus() != null ? request.getStatus() : TaskStatus.TODO);
        task.setPriority(request.getPriority());
        task.setDueDate(request.getDueDate());
        task.setAssignedTo(request.getAssignedTo());
        task.setAssignedToName(request.getAssignedToName());
        task.setCreatedAt(LocalDateTime.now());
        task.setUpdatedAt(LocalDateTime.now());
        return taskRepository.save(task);
    }

    public List<Task> getAllTasks() {
        return taskRepository.findAll();
    }

    public Task getTaskById(String id) {
        return taskRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Task not found with ID: " + id));
    }

    public List<Task> getTasksByEmployeeId(String employeeId) {
        return taskRepository.findByAssignedTo(employeeId);
    }

    public Task updateTask(String id, TaskRequest request) {
        Task task = getTaskById(id);
        task.setTitle(request.getTitle());
        task.setDescription(request.getDescription());
        if (request.getStatus() != null) {
            task.setStatus(request.getStatus());
        }
        task.setPriority(request.getPriority());
        task.setDueDate(request.getDueDate());
        task.setAssignedTo(request.getAssignedTo());
        task.setAssignedToName(request.getAssignedToName());
        task.setUpdatedAt(LocalDateTime.now());
        return taskRepository.save(task);
    }

    public Task updateTaskStatus(String id, TaskStatus status) {
        Task task = getTaskById(id);
        task.setStatus(status);
        task.setUpdatedAt(LocalDateTime.now());
        return taskRepository.save(task);
    }

    public void deleteTask(String id) {
        Task task = getTaskById(id);
        taskRepository.delete(task);
    }
}
